json_serialization_writer.go

```diff
@@ -24,40 +26,31 @@ type JsonSerializationWriter struct {
 // NewJsonSerializationWriter creates a new instance of the JsonSerializationWriter.
 func NewJsonSerializationWriter() *JsonSerializationWriter {
 	return &JsonSerializationWriter{
-		writer: make([]string, 0),
+		writer:           new(bytes.Buffer),
+		separatorIndices: make([]int, 0),
 	}
 }
-func (w *JsonSerializationWriter) writeRawValue(value string) {
-	w.writer = append(w.writer, value)
+func (w *JsonSerializationWriter) writeRawValue(value ...string) {
+	for _, v := range value {
+		w.writer.WriteString(v)
+	}
 }
 func (w *JsonSerializationWriter) writeStringValue(value string) {
+	value = strings.ReplaceAll(value, `\`, `\\`)
```

Comment:
I also cleaned this function up because it was hard to read. Functionality should be identical.

Reply:
Thanks!
this is something I haven't spent too much time on, how costly are assignments in Go?

Reply:
Not terribly expensive. Playing around with the benchmarks, it looks like the new code performs exactly the same as the existing code in terms of memory allocations and usage.